internal/repository/attestations.go

```diff
@@ -308,10 +502,104 @@ func (r *Repository) addGitHubPullRequestAttestation(ctx context.Context, signer
 	return allAttestations.Commit(r.r, commitMessage, signCommit)
 }
 
-func getGitHubClient() *github.Client {
+func indexPathToComponents(indexPath string) (string, string, string) {
+	components := strings.Split(indexPath, "/")
+
+	fromTo := strings.Split(components[len(components)-1], "-")
+	components = components[:len(components)-1] // remove last item which is from-to
+
+	base := strings.Join(components, "/") // reconstruct ref
+	from := fromTo[0]
+	to := fromTo[1]
+
+	return base, from, to
+}
+
+func getGitHubPullRequestReviewDetails(ctx context.Context, currentAttestations *attestations.Attestations, owner, repository string, pullRequestNumber int, reviewID int64) (string, string, string, error) {
+	indexPath, has := currentAttestations.GetGitHubPullRequestApprovalIndexPathForReviewID(reviewID)
+	if has {
+		base, from, to := indexPathToComponents(indexPath)
+		return base, from, to, nil
+	}
+
+	// Compute details for review, this is when the review is first created as
+	// other times we use the existing indexPath for the reviewID
+	// Note: there's the potential for a TOCTOU issue here, we may query the
+	// repo after things have moved in either branch.
+
+	client, err := getGitHubClient()
+	if err != nil {
+		return "", "", "", err
+	}
+
+	pullRequest, _, err := client.PullRequests.Get(ctx, owner, repository, pullRequestNumber)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	if _, _, err := client.PullRequests.GetReview(ctx, owner, repository, pullRequestNumber, reviewID); err != nil {
+		// testing validity of reviewID for the pull request in question
+		return "", "", "", err
+	}
+
+	baseRef := gitinterface.BranchReferenceName(*pullRequest.Base.Ref)
+
+	referenceDetails, _, err := client.Git.GetRef(ctx, owner, repository, baseRef)
+	if err != nil {
+		return "", "", "", err
+	}
+	fromID := *referenceDetails.Object.SHA // current tip of base ref
+
+	// GitHub has already computed a merge commit, use that tree ID as
+
+	commit, _, err := client.Git.GetCommit(ctx, owner, repository, *pullRequest.MergeCommitSHA)
+	if err != nil {
+		return "", "", "", err
+	}
+	toID := *commit.Tree.SHA
+
+	return baseRef, fromID, toID, nil
+}
+
+func getGitHubClient() (*github.Client, error) {
 	if githubClient == nil {
 		githubClient = github.NewClient(nil).WithAuthToken(os.Getenv("GITHUB_TOKEN"))
 	}
 
-	return githubClient
+	baseURL := os.Getenv("GITHUB_BASE_URL")
```

Comment:
I would add this in a separate change + add docs.

Reply:
https://github.com/gittuf/gittuf/pull/489